src/Tasks.h

```diff
@@ -26,25 +27,46 @@
 namespace cirrus {
 class MLTask {
   public:
-    MLTask(
-        uint64_t model_size,
-        uint64_t batch_size, uint64_t samples_per_batch,
-        uint64_t features_per_sample, uint64_t nworkers,
-        uint64_t worker_id,
-        const std::string& ps_ip,
-        uint64_t ps_port) :
-      model_size(model_size),
-      batch_size(batch_size), samples_per_batch(samples_per_batch),
-      features_per_sample(features_per_sample),
-      nworkers(nworkers), worker_id(worker_id),
-      ps_ip(ps_ip), ps_port(ps_port)
-  {}
-
-    /**
-     * Worker here is a value 0..nworkers - 1
-     */
-    void run(const Configuration& config, int worker);
-    void wait_for_start(int index, int nworkers);
+   MLTask(uint64_t model_size,
+          uint64_t batch_size,
+          uint64_t samples_per_batch,
+          uint64_t features_per_sample,
+          uint64_t nworkers,
+          uint64_t worker_id,
+          const std::string& ps_ip,
+          uint64_t ps_port)
+       : model_size(model_size),
+         batch_size(batch_size),
+         samples_per_batch(samples_per_batch),
+         features_per_sample(features_per_sample),
+         nworkers(nworkers),
+         worker_id(worker_id),
+         ps_ip(ps_ip),
+         ps_port(ps_port) {}
+   MLTask(uint64_t model_size,
```

Comment:
Same here. I don't think we need two constructors.

Reply:
Fixed